Library/VerifyGateway.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Xml;
+
+namespace Recurly
+{
+    class VerifyGateway : RecurlyEntity
+    {
+        public string GatewayCode { get; set; }
+
+        internal VerifyGateway(string gateway_code)
+        {
+            GatewayCode = gateway_code;
+        }
+
+        internal override void ReadXml(XmlTextReader reader)
+            {
+            while (reader.Read())
+            {
+              if (reader.Name == "verify" && reader.NodeType == XmlNodeType.EndElement)
+                  break;
+
+              if (reader.NodeType != XmlNodeType.Element) continue;
+
+              if (reader.Name == "gateway_code")
+                    GatewayCode = reader.ReadElementContentAsString();
+            }
+        }
+
+        internal override void WriteXml(XmlTextWriter writer)
+        {
+            writer.WriteStartElement("verify");
+
+            writer.WriteElementString("gateway_code", GatewayCode);
```

Comment:
I think that if you only include this if `GatewayCode` is not null/empty, then the conditional checks in the `Verify` method can be removed entirely.

Reply:
Following up on discussion, you'll see some adjustments. Looks much better.